developing/healthcheck/README.md

```diff
@@ -0,0 +1,44 @@
+# [Tools and Practice](../README.md) / Healthchecks
+
+## Overview
+
+In modern API services, it is not uncommon to build and rely upon Healthcheck endpoint(s). These endpoints are generally orthogonal to the operating business logic of the service, and are intended to be consumed by the Operators of the system (e.g. InfraSec practitioners).
+
+There are many ways that Healthcheck endpoints can be used in a system:
+
+* Readiness - Upon startup, has my app warmed up any internal caches that are needed before it can respond to traffic?
+* Liveness - Is this app accepting and responding to traffic at all?
+* Versioning - Is the verison of code that is deployed what it is expected to be?
+* Reachability - Can a client reach the service over the network?
+* Catalog - What are the external dependencies to which this app connects?
+* Connectivity - Can the app connect to its external dependencies successfully?
+* Quality - Does this endpoint do all its work and respond in a timely manner?
+
+Many apps will try to achieve their subset of the above goals via a single endpoint, but apps grow in sophistication it is not unheard of to tease those reponsibilities out to separate endpoints.
+
+## Authorization
+
+Frequently healthcheck endpoints are agnostic about authentication and authorization, meaning anyone that can reach the service via the network can access these endpoints. This implies that any information available in the healthcheck should not be confidential nor could be used to jeopardize the smooth running of the service.
```

Comment:
Do we want to add something about how you may need to think about DoS implications in your health check if it is unauthenticated?  Sometimes you cache the health of dependencies that are more expensive to check?

Reply:
Will do!